contracts/mixnet/src/transactions.rs

```diff
@@ -2,84 +2,27 @@
 };
 use cosmwasm_storage::ReadonlyBucket;
 use mixnet_contract::{
     Gateway, GatewayBond, IdentityKey, Layer, MixNode, MixNodeBond, RawDelegationData, StateParams,
 };
-use serde::de::DeserializeOwned;
-use serde::Serialize;
 
-const OLD_DELEGATIONS_CHUNK_SIZE: usize = 500;
+pub(crate) const OLD_DELEGATIONS_CHUNK_SIZE: usize = 500;
 pub(crate) const MINIMUM_BLOCK_AGE_FOR_REWARDING: u64 = 17280;
 
-// Looks for the total amount of delegations towards a particular node.
-// This function is used only in very specific circumstances:
-// 1. The mixnode/gateway bonds
-// 2. Some addresses start to delegate to the node
-// 3. The node unbonds
-// 4. Some of the addresses that delegated in the past have not removed the delegation yet
-// 5. The node rebonds with the same identity
-pub fn delegations<T: DeserializeOwned + Serialize>(
-    delegations_bucket: ReadonlyBucket<T>,
-) -> StdResult<Vec<(Vec<u8>, T)>> {
-    // I think it's incredibly unlikely to ever read more than that
-    // but in case we do, we should guard ourselves against possible
-    // out of memory errors (wasm contracts can only allocate at most 2MB
-    // of RAM, so we don't want to box the entire iterator)
-    let mut delegations = Vec::new();
-    // let mut total_delegation = Coin::new(0, DENOM);
-    let mut start = None;
-    loop {
-        let iterator = delegations_bucket
-            .range(start.as_deref(), None, Order::Ascending)
-            .take(OLD_DELEGATIONS_CHUNK_SIZE + 1);
-
-        let mut iterated = 0;
-
-        for result_tuple in iterator {
-            iterated += 1;
-            match result_tuple {
-                Ok((position, delegation)) => {
-                    // We might skip some values due to deserializatio errors
-                    if iterated > OLD_DELEGATIONS_CHUNK_SIZE {
-                        // we reached start of next chunk, don't process it, mark it for the next iteration of the loop
-                        start = Some(position);
-                        continue;
-                    }
-                    delegations.push((position, delegation))
-                }
-                Err(_e) => {
-                    // Skip errors
-                    continue;
-                }
-            }
-        }
-
-        if iterated <= OLD_DELEGATIONS_CHUNK_SIZE {
-            // that was the final chunk
-            break;
-        }
-    }
-
-    Ok(delegations)
-}
-
 fn total_delegations(delegations_bucket: ReadonlyBucket<RawDelegationData>) -> StdResult<Coin> {
-    match delegations(delegations_bucket) {
-        Ok(delegations) => Ok(Coin::new(
-            delegations.iter().fold(0, |acc, x| acc + x.1.amount.u128()),
-            "upunk",
-        )),
-        Err(e) => Err(e),
-    }
+    Ok(Coin::new(
+        Delegations::new(delegations_bucket)
+            .fold(0, |acc, x| acc + x.delegation_data.amount.u128()),
+        "upunk",
```

Comment:
Could we use the [`DENOM`](https://github.com/nymtech/nym/blob/develop/common/network-defaults/src/lib.rs#L69) const here?

Reply:
Looks like I've failed my own advice here :)) Will change it, thanks

Reply:
I think this one is not even yours, its adjacent :)